### ISSUE TITLE
[async branch] Query on empty table crashes client

### ISSUE THREAD

**Issue by athre0z** [suharev7/clickhouse-rs#79]
```toml
clickhouse-rs = { version = "0.2.0-alpha.3", default-features = false, features = ["async_std"] }
```

```rust
use clickhouse_rs::Pool;

async fn amain() -> std::io::Result<()> {
    let ddl = r"
        CREATE TABLE IF NOT EXISTS payment (
            customer_id  UInt32,
            amount       UInt32,
            account_name Nullable(FixedString(3))
        ) Engine=Memory";

    let database_url = "tcp://localhost:9000";
    let pool = Pool::new(database_url);

    let mut client = pool.get_handle().await?;
    client.execute(ddl).await?;
    let block = client.query("SELECT * FROM payment").fetch_all().await?;

    for row in block.rows() {
        let id: u32             = row.get("customer_id")?;
        let amount: u32         = row.get("amount")?;
        let name: Option<&str>  = row.get("account_name")?;
        println!("Found payment {}: {} {:?}", id, amount, name);
    }

    Ok(())
}

fn main() {
    async_std::task::block_on(amain()).unwrap();
}
```

crashes at

```
The application panicked (crashed).
Message:  blocks should not be empty.
Location: src/libcore/option.rs:1188
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                          (9 post panic frames hidden)
 9: core::option::expect_failed::ha19676642f985d31
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libcore/option.rs:1188
10: core::option::Option<T>::expect::h7f223c049345661a
    at <unknown source file>
11: clickhouse_rs::types::block::Block::concat::hb3631d01190f30c1
    at <unknown source file>
12: clickhouse_rs::types::query_result::QueryResult::fetch_all::{{closure}}::h5cb956ae16817174
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/clickhouse-rs-0.2.0-alpha.3/src/types/query_result/mod.rs:40
      38 │             })
      39 │             .await?;
      40 >         Ok(Block::concat(blocks.as_slice()))
      41 │     }
      42 │
13: <std::future::GenFuture<T> as core::future::future::Future>::poll::h7e4757197a3d6c15
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/future.rs:43
14: std::future::poll_with_tls_context::hcbbebb5adc01600a
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/future.rs:99
15: chrs_empty_chunk_bug::amain::{{closure}}::h0618f8e982313893
    at /Users/ath/Desktop/chrs-empty-chunk-bug/src/main.rs:16
      14 │     let mut client = pool.get_handle().await?;
      15 │     client.execute(ddl).await?;
      16 >     let block = client.query("SELECT * FROM payment").fetch_all().await?;
      17 │
      18 │     for row in block.rows() {
16: <std::future::GenFuture<T> as core::future::future::Future>::poll::hbd6938f851d171e9
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/future.rs:43
17: std::future::poll_with_tls_context::h7d159d26975d6f2c
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/future.rs:99
18: async_std::task::block_on::block_on::{{closure}}::hcf2793df5dc7a7ac
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/block_on.rs:69
      67 │         }
      68 │
      69 >         future.await
      70 │     };
      71 │
19: <std::future::GenFuture<T> as core::future::future::Future>::poll::h9514f53f0f36293e
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/future.rs:43
20: async_std::task::block_on::run::{{closure}}::h816ed4f2bfadb79c
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/block_on.rs:130
     128 │         let mut step = 0;
     129 │         loop {
     130 >             if let Poll::Ready(t) = future.as_mut().poll(cx) {
     131 │                 // Save the parker for the next invocation of `block`.
     132 │                 cache.set(Some(arc_parker));
21: std::thread::local::LocalKey<T>::try_with::hdf7835f3ad873576
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/thread/local.rs:262
22: std::thread::local::LocalKey<T>::with::hf20ce7f8c58095da
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/thread/local.rs:239
23: async_std::task::block_on::run::hae2023dda2a588bc
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/block_on.rs:119
     117 │     pin_utils::pin_mut!(future);
     118 │
     119 >     CACHE.with(|cache| {
     120 │         // Reuse a cached parker or create a new one for this invocation of `block`.
     121 │         let arc_parker: Arc<Parker> = cache.take().unwrap_or_else(|| Arc::new(Parker::new()));
24: async_std::task::block_on::block_on::{{closure}}::h291e29a105c0e435
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/block_on.rs:73
      71 │
      72 │     // Run the future as a task.
      73 >     unsafe { Task::set_current(&task, || run(future)) }
      74 │ }
      75 │
25: async_std::task::task::Task::set_current::{{closure}}::hdbfc92a8d92df19a
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/task.rs:129
     127 │                 current.set(old_task);
     128 │             }
     129 >             f()
     130 │         })
     131 │     }
26: std::thread::local::LocalKey<T>::try_with::h345ca29fc90da595
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/thread/local.rs:262
27: std::thread::local::LocalKey<T>::with::h313b34c9cb72053e
    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/thread/local.rs:239
28: async_std::task::task::Task::set_current::hd1b76895f7972498
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/task.rs:124
     122 │         F: FnOnce() -> R,
     123 │     {
     124 >         CURRENT.with(|current| {
     125 │             let old_task = current.replace(task);
     126 │             defer! {
29: async_std::task::block_on::block_on::he68527be772e060b
    at /Users/ath/.cargo/registry/src/github.com-1ecc6299db9ec823/async-std-1.4.0/src/task/block_on.rs:73
      71 │
      72 │     // Run the future as a task.
      73 >     unsafe { Task::set_current(&task, || run(future)) }
      74 │ }
      75 │
30: chrs_empty_chunk_bug::main::h411acd94b5968b0e
    at /Users/ath/Desktop/chrs-empty-chunk-bug/src/main.rs:30
      28 │ fn main() {
      29 │     color_backtrace::install();
      30 >     async_std::task::block_on(amain()).unwrap();
      31 │ }
                         (6 runtime init frames hidden)
```

**Comment by athre0z** [comment]
On first glance, it appears like we'd want to change `fetch_all` to return `Result<Option<Block<Complex>>>` instead of the current `Result<Block<Complex>>` and not call `Block::concat` when the block list is empty? 

We could also just return an empty block -- IIRC the server sends you the "sample block" even when the response is otherwise empty.

**Comment by athre0z** [comment]
Just realized that you already fixed this two weeks ago without me noticing it -- thanks! :)

**Comment by suharev7** [comment]
Fix published in [0.2.0-alpha.4](https://crates.io/crates/clickhouse-rs/0.2.0-alpha.4) and sorry for the late response.
